devito/ir/iet/visitors.py

```diff
@@ -208,7 +209,7 @@ def _gen_struct_decl(self, obj, masked=()):
             while issubclass(ctype, ctypes._Pointer):
                 ctype = ctype._type_
 
-            if not issubclass(ctype, ctypes.Structure):
+            if not issubclass(ctype, ctypes.Structure) or issubclass(ctype, NoDeclStruct):
```

Comment:
`NoDeclStruct` subclasses `ctypes.Structure`, so the `or` shouldn't be needed right?

Reply:
No, the check is inverted—we break out here if the type _is not_ a `Structure`, or if it _is_ a `NoDeclStruct`